### PR TITLE
Renames the Khalifate to Dunefolk.

### PR DESCRIPTION
**Pull request by Pentarctagon**
Based on the discussion that happened on the forums regarding the Khalifate's [faction name](https://forums.wesnoth.org/viewtopic.php?f=12&t=47643) and [race name](https://forums.wesnoth.org/viewtopic.php?f=12&t=47703), this PR renames the Khalifate to the Dunefolk.

Also, as a side note, while the poll in the race name thread was(barely) won by `Human (Dunefolk)`, I realized while creating this PR that the in-game Help entry for races uses the `plural_name` attribute of [race] rather than (as I had incorrectly assumed) a separate attribute for the Help category.  As such, I decided the in-game Help menu will instead use `Dunefolk` rather than `Human (Dunefolk)`, seeing as the latter isn't a plural name and wouldn't make sense in other places where it might be used.

### CODE REVIEW

File: data/core/units.cfg [hunk 1]
```diff
@@ -18,7 +18,7 @@
     {core/units/goblins/}
     {core/units/gryphons/}
     {core/units/humans/}
-    {core/units/khalifate/}
+    {core/units/dunefolk/}
```
Comment (ProditorMagnus) [inline data/core/units.cfg]: Move that line up, so it is in order again.

**Comment by sigurdfdragon** [comment]
I think using just 'dune' instead of 'dunefolk' for the movetypes would simpler (some of them are approaching being a mouthful), and also that just saying 'dune' is still an accurate description of the general theme of the associated movetypes.

File: data/core/macros/names.cfg [hunk 2]
```diff
@@ -139,9 +139,9 @@ centre_v=na|lla|na|ry|li|ni|ri|rae|bry
 >>
 #enddef
 
-#define KHALIFATE_NAMES
```
Comment (CelticMinstrel) [inline data/core/macros/names.cfg]: I suggest adding `KHALIFATE_NAMES` to deprecated-utils.cfg as an alias, to maintain backwards compatibility with addons.

File: data/core/units.cfg [hunk 3]
```diff
@@ -195,19 +195,19 @@ The eastern provinces of Wesnoth, known as the Clan Homelands, have a geography
...
-    # wmllint: general spellings Khalifate Khalid Khaiyal Qanas Saree Faris
+    # wmllint: general spellings Dunefolk Khalid Khaiyal Qanas Saree Faris
     # wmllint: general spellings Qatif-al-nar Rami
     # wmllint: general spelling liminal
     [race]
-        id=khalifate
-        male_name= _"race^Khalifate Human"
-        female_name= _"race+female^Khalifate Human"
-        plural_name= _"race+plural^Khalifate"
+        id=dunefolk
```
Comment (CelticMinstrel) [inline data/core/units.cfg]: This is also kinda problematic. What's really needed is for the engine to see `race=khalifate` and instead assign `race=dunefolk`... worst-case scenario, I suppose you could duplicate the entire race definition (just temporarily, until 1.14).

**Comment by Pentarctagon** [comment]
I believe that addresses everything that's been said so far.  The in-game Help also doesn't show races with no units, so that won't be a problem either.

File: changelog [hunk 4]
```diff
@@ -37,6 +37,7 @@ Version 1.13.10+dev:
    * New wesnoth.unit_types[].advances_from getter.
  * Multiplayer:
    * Dark Forecast: Fixed broken faction and leader selection.
+   * Rename the Khalifate to the Dunefolk.
```
Comment (CelticMinstrel) [inline changelog]: I'd remove the second "the"; it sounds a little awkward to me.

File: data/tools/wmllint [hunk 5]
```diff
@@ -835,10 +835,10 @@ linechanges = (
         
         # renamed khalifate to dunefolk
         ("id=khalifate", "id=dunefolk"),
```
Comment (CelticMinstrel) [inline data/tools/wmllint]: Shouldn't this be `race=` rather than `id=`?

Comment (Pentarctagon) [reply data/tools/wmllint]: I do have `race=` a few lines below.

File: changelog (hunk 4 above)
Comment (Pentarctagon) [reply changelog]: Fair enough.

**Comment by sigurdfdragon** [comment]
I'm fine with how the plural race name is handled.

As we're doing some backward compatibility for a time, should we also be doing that for the faction file includes?

Say this file {multiplayer/factions/khalifate-default.cfg} still existing, but containing only a file include of
{multiplayer/factions/dunefolk-default.cfg} as its contents?

**Comment by Pentarctagon** [comment]
I've added that as well.

**Comment by sigurdfdragon** [comment]
Maybe wmllint entries for the era id's would also be good?
era_khalifate -> era_dunefolk
era_khalifate_heroes -> era_dunefolk_heroes

Also, I think a descriptive commit message is in order here that also contains a forum link. Something along the lines of:
Part of an effort to give the faction a more Wesnoth feel & address concerns.
From forum discussion https://forums.wesnoth.org/viewtopic.php?f=12&t=47643

& while you're editing the commit message, a title of:
MP: Rename the Khalifate to Dunefolk
would fit convention a bit better.

Other than that, looks good.
If there are no further issues or discussion, I can merge this sometime Thursday or Friday

**Comment by Pentarctagon** [comment]
Alright, sounds like it should be good to go then.

**Pull request merged**